src/axom/slic/CMakeLists.txt

```diff
@@ -75,6 +75,11 @@ blt_add_library(NAME        slic
                 OBJECT      TRUE
                 )
 
+# If we're compiling as a dynamic-link library, define AXOM_EXPORTS so we export symbols.
+if(BUILD_SHARED_LIBS)
+    target_compile_definitions(slic PRIVATE AXOM_EXPORTS)
+endif()
```

Comment:
My first try was to put AXOM_EXPORTS in a list and pass that to DEFINES argument of blt_add_library.  That, however, adds the symbols as PUBLIC.  So AXOM_EXPORTS got propagated up through axom.dll to the mesh_tester executable, which caused a linking error.

@white238 , should we maybe add a DEFINES_PRIVATE argument to blt_add_library for this purpose, or is the need too specialized?

Reply:
https://llnl-blt.readthedocs.io/en/develop/api/target_properties.html#blt-add-target-definitions

This takes a scope.

Reply:
for context (which I think @agcapps has, but just wanted to share)

for each lib, cmake provides an automatic export define that matches the lib name. 
if the lib name doesn't match the exports in the header, you need to provide your own define at compile time + you *don't* want that define to to propagate to calling code.